pkg/contracts/solana/gateway.go

```diff
@@ -17,6 +17,10 @@ const (
 	// AccountsNumberOfDeposit is the number of accounts required for Solana gateway deposit instruction
 	// [signer, pda, system_program]
 	AccountsNumDeposit = 3
+
+	// AccountsNumberOfDeposit is the number of accounts required for Solana gateway deposit spl instruction
+	// [signer, pda, whitelist_entry, mint_account, token_program, from, to]
+	AccountsNumberDepositSPL = 7
```

Comment:
Can we unexport these constants and leave the implementation within this package?

Reply:
Before doing further changes here, I might think this is a value we can get from generated code in the Solana repo https://github.com/zeta-chain/protocol-contracts-solana/pull/52

Reply:
@skosito other than this I think this might make sense to try moving in this PR the parsing logic for the inbound in the Solana package rather than in the ZetaClient